src/OpenLoco/Windows/PromptBrowseWindow.cpp

```diff
@@ -775,26 +831,103 @@ namespace OpenLoco::Ui::PromptBrowse
     }
 
     // 0x00446574
-    static void sub_446574(Ui::window* window)
+    static void processFileForLoadSave(window* self)
     {
-        if (*_type == browse_type::save)
+        if (_type == browse_type::save)
         {
             if (filenameContainsInvalidChars())
             {
                 Windows::showError(StringIds::error_invalid_filename);
+                return;
             }
+
+            // Create full path to target file.
+            fs::path path = fs::path(&_directory[0]) / std::string(&_text_input_buffer[0]);
+
+            // Append extension to filename.
+            if (_fileType == browse_file_type::saved_game)
+                path += ".SV5";
             else
+                path += ".SC5";
+
+            // Does the file already exist?
+            if (fs::exists(path))
             {
-                registers regs;
-                regs.esi = (int32_t)window;
-                call(0x00446598, regs);
+                // Copy directory and filename to buffer.
+                char* buffer_2039 = const_cast<char*>(StringManager::getString(StringIds::buffer_2039));
+                strncpy(&buffer_2039[0], path.c_str(), 512);
```

Comment:
Some of the CI jobs are complaining about going from `fs::value_type*` to `const char*` here, whilst others are fine. Any ideas how best to resolve this? I'd rather avoid casting, if possible.

Reply:
Will be wchar_t on Windows and char on posix. You'll need to put it through some sort of conversion prior to this i think.